addons/mail/static/src/components/chatter/chatter_tests.js

```diff
@@ -453,6 +462,346 @@ QUnit.test('do not post message with "Enter" keyboard shortcut', async function
+        threadId: 10,
+        threadModel: 'res.fake',
+    });
+    await this.createChatterComponent({ chatter });
+    await afterNextRender(() =>
+        document.querySelector(`.o_ChatterTopbar_buttonSendMessage`).click()
+    );
+    const checkboxUnchecked = document.querySelector('.o_ComposerSuggestedRecipient:not([data-partner-id]) input[type=checkbox]');
+    assert.notOk(
+        checkboxUnchecked.checked,
+        "suggested recipient without a partner should not be checked by default",
+    );
+});
+
+QUnit.test("Suggested recipient with partner checked by default", async function (assert) {
+    assert.expect(1);
+
+    this.data['res.partner'].records.push({
+        display_name: "John Jane",
+        email: "john@jane.be",
+        id: 100,
+    });
+    this.data['res.fake'].records.push({
+        id: 10,
+        partner_ids: [100],
+    });
+    await this.start();
+    const chatter = this.env.models['mail.chatter'].create({
+        threadId: 10,
+        threadModel: 'res.fake',
+    });
+    await this.createChatterComponent({ chatter });
+    await afterNextRender(() =>
+        document.querySelector(`.o_ChatterTopbar_buttonSendMessage`).click()
+    );
+    const checkboxChecked = document.querySelector('.o_ComposerSuggestedRecipient[data-partner-id="100"] input[type=checkbox]');
+    assert.ok(
+        checkboxChecked.checked,
+        "suggested recipient with a partner should be checked by default.",
+    );
+});
+
+QUnit.test("suggested recipients list display a \"show more\" button when >= 4 are available", async function (assert) {
+    assert.expect(1);
+
+    this.data['res.partner'].records.push({
+        display_name: "John Jane",
+        email: "john@jane.be",
+        id: 100,
+    });
+    this.data['res.partner'].records.push({
+        display_name: "Jack Jone",
+        email: "jack@jone.be",
+        id: 1000,
+    });
+    this.data['res.partner'].records.push({
+        display_name: "jolly Roger",
+        email: "Roger@skullflag.com",
+        id: 1001,
+    });
+    this.data['res.partner'].records.push({
+        display_name: "jack sparrow",
+        email: "jsparrow@blackpearl.bb",
+        id: 1002,
+    });
+    this.data['res.fake'].records.push({
+        id: 10,
+        partner_ids: [100, 1000, 1001, 1002],
+    });
+    await this.start ();
+    const chatter = this.env.models['mail.chatter'].create({
+        threadId: 10,
+        threadModel: 'res.fake',
+    });
+    await this.createChatterComponent({ chatter });
+
+    await afterNextRender(() =>
+        document.querySelector(`.o_ChatterTopbar_buttonSendMessage`).click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ComposerSuggestedRecipientList_showMore',
+        "suggested recipient list should contain a \"show more\" button."
+    );
+});
+
+QUnit.test("suggested recipients list display all recipients when \"show more\" button is clicked", async function (assert) {
```

Comment:
_"more than 3 suggested recipients: show all of them on click 'show more' button"_